File: problem_657_robot_return_to_origin/return_to_origin.py
#!/usr/bin/python3
class Solution:
    def judgeCircle(self, moves):
        x = 0
        y = 0

        for move in moves:
            if move == 'U': y+=1
            if move == 'D': y-=1
            if move == 'R': x+=1
            if move == 'L': x-=1
        
        if x == 0 and y == 0: return True
        else: return False

def main():
    print(obj.judgeCircle("UDU"))

# Start program
if __name__ == "__main__":
    obj = Solution()
    main()
    

File: problem_657_robot_return_to_origin/return_to_origin.cpp
#include<iostream>

using namespace std;

bool judgeCircle(string moves) {
        int x = 0;
        int y = 0;
    
        for (int i = 0; i < moves.size(); i++) {
		if (moves[i] == 'U') {y++;}
        if (moves[i] == 'D') {y--;}
        if (moves[i] == 'R') {x++;}
	    if (moves[i] == 'L') {x--;}
        }   
        
        
        if (x == 0 && y == 0) {return true;}
        else {return false;}
        
    }

int main(){

    cout<<  judgeCircle("UDU") <<endl;
    return 0;

}